### PR TITLE
Season refactor

### PR DESCRIPTION
**Pull request by sciencefixion**
This is my first pass at a refactor. Obviously, I'm over on some of my helper method lines, but l figured I might take another pass to try to incorporate outside methods tomorrow. Let me know what you think. @gabichuelas @evilaspaas1 @rrabinovitch

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -160,55 +160,47 @@ def lowest_scoring_home_team
...
+  def season_game_teams(season)
+    season_game_ids = season_games(season).map { |game| game.game_id }
+    game_teams.find_all { |game| season_game_ids.include?(game.game_id) }
+  end
 
-    game_teams_by_coach = season_game_teams.group_by do |game_team|
-      game_team.head_coach
-    end
+  def coach_win_percentage(season)
+    game_teams_by_coach = season_game_teams(season).group_by { |game_team| game_team.head_coach }
```
Comment (rrabinovitch) [inline lib/stat_tracker.rb:201]: given SRP, this calculation should probably be separated into another helper method

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -160,55 +160,47 @@ def lowest_scoring_home_team
...
 
     coach_games_and_wins = Hash.new { |h,k| h[k] = Hash.new(0) }
```
Comment (rrabinovitch) [inline lib/stat_tracker.rb:203]: does this setup serve a different purpose than just doing `coach_games_and_wins = Hash.new(0)`? I think giving more specific block variable names to the `h`s and `k`s would be helpful. (it's late so it's very possible I'm missing something :) )

also another section that should probably be a separate helper method

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -218,21 +210,12 @@ def worst_coach(season)
...
-    season_game_ids = season_games(season).map do |game|
-      game.game_id
-    end
-
-    season_game_teams = game_teams.find_all do |game|
-      season_game_ids.include?(game.game_id)
-    end
-
+  def team_accuracy(season)
     season_team_accuracy = Hash.new { |h,k| h[k] = Hash.new(0) }
```
Comment (rrabinovitch) [inline lib/stat_tracker.rb:217]: another one to hopefully separate out

**Review by rrabinovitch — CHANGES_REQUESTED** [review]
awesome job at breaking your methods down! one more note in addition to the comments I left about further breaking down some of your helper methods: most of the methods using `min_by` and `max_by` go past the 80 character/line limit (`most_accurate_team` - just barely so maybe it's fine as is, `least_accurate_team`, `most_tackles`, `fewest_tackles`). my understanding is that we should only use the oneline enumerable structure when it fits within that character limit. and since the methods are now pretty pared down, are concise, and they follow SRP, I think it's okay to expand those enumerables into the longer format so that they're not pushing past that line length too much

File: lib/stat_tracker.rb (hunk 2 above)
Comment (gabichuelas) [reply lib/stat_tracker.rb:203]: the block isn't needed here, it could just be `coach_games_and_wins = Hash.new(0)`

Comment (gabichuelas) [reply lib/stat_tracker.rb:203]: After meeting, realized Dorion is trying to set default values to be _another_ Hash with default values at 0.

**Comment by gabichuelas** [comment]
Making executive decision to approve this PR so we can have progress on master, and then Dorion / us can keep refactoring further.

**Pull request merged**